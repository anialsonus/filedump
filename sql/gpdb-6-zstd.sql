-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, compresstype=zstd)
\! pg_filedump -z -M -k -T zstd test/ao.zstd.gp6

-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, orientation=column, compresstype=zstd)
\! pg_filedump -z -M -k -O column -T zstd test/co.zstd.gp6
