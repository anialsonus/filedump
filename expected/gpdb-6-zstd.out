-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, compresstype=zstd)
\! pg_filedump -z -M -k -T zstd test/ao.zstd.gp6

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: test/ao.zstd.gp6
* Options used: -z -M -k -T zstd
*******************************************************************

 Initial Bytes Read = 512

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0xff68d373)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 692 compressedLen: 226
 firstRowNumber: 1
Content checksum valid (checksum: 0x0012004e)

***** Block    0 End (Total Block length:  256) ******************
 Initial Bytes Read = 256

***** Block    1 Start ***( Position:  256) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x249741ce)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 692 compressedLen: 226
 firstRowNumber: 101
Content checksum valid (checksum: 0x41d7e725)

***** Block    1 End (Total Block length:  256) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, orientation=column, compresstype=zstd)
\! pg_filedump -z -M -k -O column -T zstd test/co.zstd.gp6

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: test/co.zstd.gp6
* Options used: -z -M -k -O column -T zstd
*******************************************************************

 Initial Bytes Read = 240

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0xb2a7c6ee)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 168 compressedLen: 92
 firstRowNumber: 1
Content checksum valid (checksum: 0x6adaf236)

	<AO Content Info> -----
	 Decompressing using "ZSTD" with compressionLevel: 0
	 DatumStreamVersion_Original  flags: 0
	 ndatum: 38 nullSize: 0 Size: 152
***** Block    0 End (Total Block length:  120) ******************
 Initial Bytes Read = 120

***** Block    1 Start ***( Position:  120) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x531e173f)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 168 compressedLen: 92
 firstRowNumber: 101
Content checksum valid (checksum: 0x098081d2)

	<AO Content Info> -----
	 Decompressing using "ZSTD" with compressionLevel: 0
	 DatumStreamVersion_Original  flags: 0
	 ndatum: 38 nullSize: 0 Size: 152
***** Block    1 End (Total Block length:  120) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
